uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Reminder.php

```diff
@@ -131,14 +131,19 @@ private static function _extractPreviousAssignee(&$assignees, $previousAssigneeI
    * Gets the full list of the recipients of the reminder email
    * Makes sure there are no duplicates or no empty emails in the list
    *
-   * @param array $activityContacts
+   * @param array $contacts
    * @param array $previousAssignee
    * @return array
    */
-  private static function _reminderRecipients($activityContacts, $previousAssignee) {
-    return array_filter(array_unique(array_merge(
-                            $activityContacts['assignees']['emails'], $activityContacts['source']['emails'], [$previousAssignee['email']]
-    )));
+  private static function _reminderRecipients($contacts, $previousAssignee) {
+    $hasSources = !empty($contacts['source']['emails']);
```

Comment:
Good catch with the missing `'email'` index

The `$assignees = $contacts['assignees']['emails'] ?: [];` is nice and short, but with errors on it can lead to an undefined notices if 'emails' is not set. I know it's not so terrible, but I would prefer not to have any notices in development because if we have 1000 of them maybe one will get through that's actually a bug.